presto-tests/src/test/java/com/facebook/presto/tests/TestTpchDistributedStats.java

```diff
@@ -0,0 +1,131 @@
+ * limitations under the License.
+ */
+package com.facebook.presto.tests;
+
+import com.facebook.presto.tests.statistics.StatisticsAssertion;
+import com.facebook.presto.tpch.ColumnNaming;
+import com.google.common.collect.ImmutableMap;
+import io.airlift.tpch.TpchTable;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.tests.statistics.Metric.OUTPUT_ROW_COUNT;
+import static com.facebook.presto.tests.statistics.MetricComparisonStrategies.absoluteError;
+import static com.facebook.presto.tests.statistics.MetricComparisonStrategies.defaultTolerance;
+import static com.facebook.presto.tests.statistics.MetricComparisonStrategies.noError;
+import static com.facebook.presto.tests.statistics.MetricComparisonStrategies.relativeError;
+import static com.facebook.presto.tests.tpch.TpchQueryRunner.createQueryRunnerWithoutCatalogs;
+import static java.util.Collections.emptyMap;
+
+public class TestTpchDistributedStats
+{
+    private StatisticsAssertion statisticsAssertion;
+
+    @BeforeClass
+    public void setup()
+            throws Exception
+    {
+        DistributedQueryRunner runner = createQueryRunnerWithoutCatalogs(emptyMap(), emptyMap());
+        runner.createCatalog(
+                "tpch",
+                "tpch",
+                ImmutableMap.of("tpch.column-naming", ColumnNaming.STANDARD.name()));
+        statisticsAssertion = new StatisticsAssertion(runner);
+    }
+
+    @AfterClass
+    public void tearDown()
+    {
+        statisticsAssertion.close();
+        statisticsAssertion = null;
+    }
+
+    @Test
+    public void testTableScanStats()
+    {
+        TpchTable.getTables()
+                .forEach(table -> statisticsAssertion.check("SELECT * FROM " + table.getTableName())
+                        //TODO use noError
+                        .estimate(OUTPUT_ROW_COUNT, defaultTolerance()));
+    }
+
+    @Test
+    public void testFilter()
+    {
+        statisticsAssertion.check("SELECT * FROM lineitem WHERE l_shipdate <= DATE '1998-12-01' - INTERVAL '90' DAY")
+                //TODO fix estimates
+                .estimate(OUTPUT_ROW_COUNT, relativeError(-.55, -.45));
+    }
+
+    @Test
+    public void testJoin()
+    {
+        statisticsAssertion.check("SELECT * FROM  part, partsupp WHERE p_partkey = ps_partkey")
+                .estimate(OUTPUT_ROW_COUNT, relativeError(.95, 1.05));
+    }
+
+    @Test
+    public void testSetOperations()
+    {
+        statisticsAssertion.check("SELECT * FROM nation UNION SELECT * FROM nation")
+                .noEstimate(OUTPUT_ROW_COUNT);
+
+        statisticsAssertion.check("SELECT * FROM nation INTERSECT SELECT * FROM nation")
+                .noEstimate(OUTPUT_ROW_COUNT);
+
+        statisticsAssertion.check("SELECT * FROM nation EXCEPT SELECT * FROM nation")
+                .noEstimate(OUTPUT_ROW_COUNT);
```

Comment:
Run assertion and check there isn't any estimate? Why do we have it in that form?

Reply:
No support for `EXCEPT` in stats estimation math I think.